src/variables.jl

```diff
@@ -3,6 +3,24 @@
 #  License, v. 2.0. If a copy of the MPL was not distributed with this
 #  file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+abstract type AbstractVariable end
+
+mutable struct VariableInfo{S, T, U, V}
+    haslb::Bool
+    lowerbound::S
+    hasub::Bool
+    upperbound::T
+    hasfix::Bool
+    fixedvalue::U
+    hasstart::Bool
+    start::V
+    binary::Bool
+    integer::Bool
+end
+
+struct ScalarVariable{S, T, U, V}
```

Comment:
`JuMP.Variable` needs to be renamed to `JuMP.VariableRef` before we can have an object called `ScalarVariable` with this meaning.

Reply:
(If you do that, make it a separate PR and then update this one afterwards.)

Reply:
I have rebase the PR on master, it is now ready for review :)